### ISSUE TITLE
Can't build a project on macOS after XamlX update [11.2 alpha]

### ISSUE THREAD

**Issue by BAndysc** [AvaloniaUI/Avalonia#15436]
### Describe the bug

After this commit https://github.com/AvaloniaUI/Avalonia/commit/d5ca06ff2432dcc4671a4ed795a5ce6a5cdadddc, my project doesn't build on macOS
```
MSBUILD : Avalonia error AVLN9999: Failed to resolve assembly: 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' [/WoWDatabaseEditor/Modules/WDE.QuestChainEditor/WDE.QuestChainEditor.csproj]
    217 Warning(s)
    1 Error(s)
```

### To Reproduce

1. `git clone https://github.com/BAndysc/avalonia-bugs-repro`
2. `git checkout bug_presentation_framework`
3. `dotnet build`

### Avalonia version

11.2.999-cibuild0047389-alpha

### OS

macOS

**Comment by timunie** [comment]
`PresentationFramework` sounds like WPF code somewhere

**Comment by timunie** [comment]
maybe this package? https://github.com/BAndysc/WoWDatabaseEditor/blob/5d8b410b81cbc61f7deee5fc5fa1ffc0bcb5f527/WDE.QuestChainEditor/WDE.QuestChainEditor.csproj#L19

**Comment by BAndysc** [comment]
Good catch, I had an old project with the same name in some old folder (Wasn't used at all, neither by the solution nor referenced by csproj), removed it to make sure (and pushed), but it still doesn't help. Extremely weird and it doesn't happen before the linked commit.

**Comment by maxkatz6** [comment]
I assume types from this namespace are no used in your Avalonia XAML, right? 
Ideally, we shouldn't try to resolve these.

**Comment by BAndysc** [comment]
They are not.

---

Eventually I managed to workaround this, in general the problem was [GraphX](https://www.nuget.org/packages/GraphX) package. I don't really understand how it works, but this package has both platform agnostic code and some WPF code, which are later bundled into a single nuget package. But I am only using the platform independent parts of the library and it worked fine before, then it stopped working with the error seen above. I ended up manually adding the sources of this library to my solution and including only the two projects (platform independent) that I need.

I have tried to reproduce it with a clean Avalonia project (with the same nightly build) - I have referenced this nuget package, but it still built fine. Maybe this is some deep weird configuration in my solution, no idea. I guess for now the issue can be closed, as I can't really tell what the problem is.

**Comment by maxkatz6** [comment]
> I don't really understand how it works, but this package has both platform agnostic code and some WPF code

They have netcoreapp3.1 target, which technically can reference wpf assemblies, if used.

> I have tried to reproduce it with a clean Avalonia project (with the same nightly build) - I have referenced this nuget package, but it still built fine. 

Try to use any of the types available in `GraphX.Core.Controls.dll` package. I.e. something like `CustomHelper.IsIntegerInput("123123")` in your C# code. Without doing anything like this in XAML.
C# code won't fail compilation on that, neither should XAML. 

> I guess for now the issue can be closed, as I can't really tell what the problem is.

Even with a lower priority, I would keep it open. This issue might be annoying for more than one project.

<img width="770" alt="Screenshot 2024-04-18 at 4 06 00 PM" src="https://github.com/AvaloniaUI/Avalonia/assets/3163374/dce5e50d-d0c7-4fc7-a407-e0d11b3b5ee8">

**Comment by BAndysc** [comment]
@maxkatz6 
Alright, I was able to produce a minimal repro: https://github.com/BAndysc/avalonia-bugs-repro, branch `bug_presentation_framework`.

I have no idea why, but these two packages (the first one is my port)

```
        <PackageReference Include="NodifyAvalonia" Version="5.2.0.1" />
        <PackageReference Include="GraphX" Version="3.0.0" />
```

interfere with each other beginning with the commit from the original issue. It works fine with previous Avalonia versions.

If I remove either of those packages, it builds fine. Also, no code is needed at all, just include those two packages and the latest Avalonia nightly. `NodifyAvalonia` references Avalonia version 11.1.0-beta1, which initially I thought might be the problem, but no - I have changed it to the very same Avalonia 11.2-alpha version and still get the `PresentationFramework` error when building. Now I am curious why it happens...
